source/framework/vx_array.c

```diff
@@ -22,6 +22,182 @@ static void ownInitArrayObject(
+ * \returns VX_SUCCESS if it is, otherwise another error code.
+ *
+ */
+static vx_status isArrayCopyable(vx_array input, vx_array output)
+{
+    vx_status status = (vx_status)VX_SUCCESS;
+    tivx_obj_desc_array_t *ip_obj_desc = (tivx_obj_desc_array_t *)input->base.obj_desc;
+    tivx_obj_desc_array_t *op_obj_desc = (tivx_obj_desc_array_t *)output->base.obj_desc;
+    if (vx_false_e == ownIsValidInputAndOutputArrays(input, output))
+    {
+        status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+    }
+    else if (vx_true_e == output->base.is_virtual)
+    {
+        /* Either output type must be invalid or types must match*/
+        if ((VX_TYPE_INVALID != op_obj_desc->item_type) &&
+            (ip_obj_desc->item_type != op_obj_desc->item_type))
+        {
+            status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+        }
+        /* Either output capacity must be zero or at least as large as input capacity */
+        if ((0 != op_obj_desc->capacity) &&
+             (ip_obj_desc->capacity != op_obj_desc->capacity))
+        {
+            status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+        }
+    }
+    else if (op_obj_desc->item_type != ip_obj_desc->item_type)
+    {
+        /* Types must match */
+        status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+    }
+    else if (op_obj_desc->capacity < ip_obj_desc->capacity)
+    {
+        /* Output must have sufficient capacity to hold input */
+        status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+    }
+    else
+    {
+        op_obj_desc->item_type = ip_obj_desc->item_type;
+        op_obj_desc->item_size = ip_obj_desc->item_size;
+        op_obj_desc->capacity = ip_obj_desc->capacity;
+    }
+    return status;
+}
+
+/*! \brief This function is called to find out if it is OK to swap the input with the output.
+ * Both item type and capacity must be the same.
+ * \returns VX_SUCCESS if it is, otherwise another error code.
+ *
+ */
+static vx_status isArraySwapable(vx_array input, vx_array output)
+{
+    vx_status status = (vx_status)VX_SUCCESS;
+    tivx_obj_desc_array_t *ip_obj_desc = (tivx_obj_desc_array_t *)input->base.obj_desc;
+    tivx_obj_desc_array_t *op_obj_desc = (tivx_obj_desc_array_t *)output->base.obj_desc;
+    if (vx_false_e == ownIsValidInputAndOutputArrays(input, output))
+    {
+        status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+    }
+    else if (op_obj_desc->item_type != ip_obj_desc->item_type)
+    {
+        /* Types must match */
+        status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+    }
+    else if (op_obj_desc->capacity != ip_obj_desc->capacity)
+    {
+        /* Output must have same capacity as input */
+        status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+    }
+    return status;
+}
+
+/*! \brief Copy input to output
+ * The input must be copyable to the output; checks done already.
+ * Note that locking a reference actually locks the context, so we only lock
+ * one reference!
+ */
+static vx_status copyArray(vx_array input, vx_array output)
```

Comment:
GENERIC: This looks like it can also be done in a generic way across most references similar to ownAllocReferenceBufferGeneric.  The generic function can be in vx_reference.c

Reply:
tivxIsReferenceMetaFormatEqual
vx_lut copyable --> ok
vx_matrix --> origin and pattern is missing
vx_convolution --> mo meta comparision for convolution
vx_distrbution --> ok
vx_scalar --> ok
vx_tensor --> not all data checked like scaling_divisor_fixed_point_position
vx_pyramid --> specific checks if it is virtual
vx_image --> also too specific
vx_objarray --> ok, but questions in the code

swapable only two objects have a specific funtion
- vx_image --> cannot be use because of the subimage checked
- vx_array --> ok

copy function: (ownCopyReferenceGeneric)
vx_distrib: done
vx_array : done
vx_matrix: done
vx_lut: done
scalar: not needed
vx_tensor: done
vx_convolution: done
vx_distrbution: done
vx_pyramid: not applicable
vx_image / raw_image : not applicable
vx_objarray: not applicable

swap function:
vx_lut: done
vx_distrbution: done
vx_matrix: done
vx_array: done
vx:tensor: mixed with move, not possible
vx_convolution: done
vx_pyramid: not applicable
vx_image / raw_image : not applicable
vx_objarray: not applicable